src/Resource.ts

```diff
@@ -17,15 +21,47 @@ export enum Method {
 const isJson = (response: AxiosResponse<unknown>): boolean =>
   (response.headers["content-type"] || "").includes("application/json");
 
-const convertAxiosErrorToOnfidoError = (error: any): OnfidoError => {
-  if (error.response) {
-    // Received a 4XX or 5XX response.
-    const response: AxiosResponse = error.response;
-    return OnfidoApiError.fromResponse(response.data, response.status);
-  } else if (error.message) {
-    return new OnfidoError(error.message);
-  } else {
-    return new OnfidoError("An unknown error occurred making the request");
+const readFullStream = (stream: Readable): Promise<unknown> =>
+  new Promise((resolve): void => {
+    let all = "";
+
+    stream.on("data", data => (all += data));
+    stream.on("error", () => resolve("An error occurred reading the response"));
+    stream.on("end", () => {
+      try {
+        resolve(JSON.parse(all));
+      } catch {
+        resolve(all);
```

Comment:
In case the error isn't json e.g. some server errors might just return text instead. I suppose I could check the content type instead but that's not always 100% reliable.

Reply:
I will add a comment